packages/aserto/test/test_directory.py

```diff
@@ -0,0 +1,149 @@
+        client=directory_client,
+        obj_1=obj_1,
+        obj_2=obj_2,
+        obj_3=obj_3,
+        relation_1=relation_1,
+        relation_2=relation_2,
+    )
+
+    relations = directory_client.get_relations(page=PaginationRequest(size=30)).results
+    for rel in relations:
+        directory_client.delete_relation(
+            subject_type=rel.subject.type,
+            subject_key=rel.subject.type,
+            object_type=rel.object.type,
+            object_key=rel.object.key,
+            relation_type=rel.relation,
+        )
+
+    objects = directory_client.get_objects(page=PaginationRequest(size=30)).results
+    for obj in objects:
+        directory_client.delete_object(key=obj.key, type=obj.type)
+
+
+def test_delete_object(directory):
+    obj_to_delete = directory.client.get_object(key=directory.obj_1.key, type=directory.obj_1.type)
+    directory.client.delete_object(key=obj_to_delete.key, type=obj_to_delete.type)
+    with pytest.raises(NotFoundError):
+        directory.client.get_object(key=directory.obj_1.key, type=directory.obj_1.type)
+
+
+def test_get_object(directory):
+    obj = directory.client.get_object(key=directory.obj_1.key, type=directory.obj_1.type)
+    assert obj.key == directory.obj_1.key
+    assert obj.type == directory.obj_1.type
+    assert obj.display_name == directory.obj_1.display_name
+
+
+def test_object_not_found(directory):
+    key = uuid.uuid4().hex
+    with pytest.raises(NotFoundError):
+        directory.client.get_object(key=key, type="user")
+
+
+def test_object_invalid_arg(directory):
+    with pytest.raises(TypeError):
+        directory.client.get_object(key=directory.obj_1.key)
+
+
+def test_get_objects_by_type(directory):
+    objs = directory.client.get_objects(
+        object_type=directory.obj_1.type, page=PaginationRequest(size=10)
+    ).results
+    print(objs)
+    assert directory.obj_1 in objs
+    assert directory.obj_1.type == directory.obj_3.type
+    assert directory.obj_1.type != directory.obj_2.type
+    assert directory.obj_3 in objs
+    # assert directory.obj_2 not in objs
+
+
+def test_get_objects(directory):
+    objs = directory.client.get_objects(page=PaginationRequest(size=10)).results
+    assert directory.obj_1 in objs
+    assert directory.obj_2 in objs
+    assert directory.obj_3 in objs
+
+
+def test_get_objects_many(directory):
+    objs = directory.client.get_objects_many(
+        objects=[
+            ObjectIdentifier(key=directory.obj_1.key, type=directory.obj_1.type),
+            ObjectIdentifier(key=directory.obj_2.key, type=directory.obj_2.type),
+        ]
+    )
+    assert directory.obj_1 in objs
+    assert directory.obj_2 in objs
+    assert directory.obj_3 not in objs
+
+
+def test_set_object(directory):
+    key = uuid.uuid4().hex
+
+    obj = directory.client.set_object(Object(key=key, type="user", display_name="test user"))
+
+    assert obj.key == key
+    assert obj.type == "user"
+    assert obj.display_name == "test user"
```

Comment:
Not sure how useful this test is because if `set_object` didn't work, the test would fail at the fixture level before this function is even called.
What you _can_ do though, is test that you are able to modify an existing object.